### PR TITLE
Add DirectxTK for windows

### PR DESCRIPTION
**Pull request by yamashi**

### CODE REVIEW

File: packages/d/directxtk/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package("DirectXTK")
+
+    set_homepage("https://github.com/Tessil/hopscotch-map")
+    set_description("This package contains the \"DirectX Tool Kit\", a collection of helper classes for writing Direct3D 11 C++ code for Universal Windows Platform (UWP) apps for Windows 10, Xbox One, and Win32 desktop applications for Windows 7 Service Pack 1 or later.")
+
+    set_urls("https://github.com/microsoft/DirectXTK/archive/$(version).zip",
+             "https://github.com/microsoft/DirectXTK.git",
+             {version = function (version)
+                local versions = {
+                    ["20.9.0"]="sept2020"
```
Comment (waruqi) [inline packages/d/directxtk/xmake.lua:10]: fix code style: `] = "sept2020"

File: packages/d/directxtk/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                    ["20.9.0"] = "sept2020"
+                }
+
+                return versions[version]
+            end})
+
+    add_versions("20.9.0", "9d5131243bf3e33db2e3a968720d860abdcbbe7cb037c2cb5dd06046d439ed09")
+
+    on_install("windows", function (package)
+        local configs = { "-DCMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION=" .. package:config("vs_sdkver") }
```
Comment (waruqi) [inline packages/d/directxtk/xmake.lua]:
```lua
local vs_sdkver = get_config("vs_sdkver")
if vs_sdkver then
    table.insert(configs, "-DCMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION=" .. vs_sdkver)
end
```

**Comment by yamashi** [comment]
Please squash this disaster ^^

File: packages/d/directxtk/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    for match in (s..delimiter):gmatch("(.-)"..delimiter) do
+        table.insert(result, match);
+    end
+    return result;
+end
+
+
+package("DirectXTK")
+
+    set_homepage("https://github.com/Tessil/hopscotch-map")
```
Comment (SirLynix) [inline packages/d/directxtk/xmake.lua:12]: Wrong homepage

File: packages/d/directxtk/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            end})
+
+    add_versions("20.9.0", "9d5131243bf3e33db2e3a968720d860abdcbbe7cb037c2cb5dd06046d439ed09")
+
+    on_install("windows", function (package)
+        local configs = {}
+        
+        local vs_sdkver = get_config("vs_sdkver")
+        if vs_sdkver then
+            split_version = {};
```
Comment (SirLynix) [inline packages/d/directxtk/xmake.lua:32]: Using Lua patterns would be an improvement, like `string.match("10.0.19041.0", "%d+%.%d+%.(%d+)%.?%d*")`

**Review by SirLynix — CHANGES_REQUESTED** [review]

File: packages/d/directxtk/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    on_install("windows", function (package)
+        local configs = {}
+        
+        local vs_sdkver = get_config("vs_sdkver")
+        if vs_sdkver then
+            local build_ver = string.match(vs_sdkver, "%d+%.%d+%.(%d+)%.?%d*")
+
+            assert(tonumber(build_ver) >= 19041, "DirectXTK requires Windows SDK to be at least 10.0.19041.0")
+
```
Comment (waruqi) [inline packages/d/directxtk/xmake.lua:36]: Please remove empty lines

File: packages/d/directxtk/xmake.lua [hunk 6]
```diff
@@ -6,7 +6,6 @@ function split(s, delimiter)
     return result;
 end
 
-
```
Comment (waruqi) [inline packages/d/directxtk/xmake.lua]: please remove unused split function

**Comment by waruqi** [comment]
In order to avoid too long commit history affecting git pull time, please merge all commit to a single commit.

**Comment by yamashi** [comment]
> 
> 
> In order to avoid too long commit history affecting git pull time, please merge all commit to a single commit.

You can do a squash merge it will make a single commit

**Comment by waruqi** [comment]
I have used `git merge --squash directxtk; git commit -a -m "merge directxtk"` to do this squash merge.

But it will lose contributor information。

**Comment by yamashi** [comment]
Github itself gives you the option to squash merge, you don't need to do it yourself

**Pull request closed without merging**